test/client/dio_interceptor_test.dart

```diff
@@ -0,0 +1,197 @@
+import 'dart:io';
+
+import 'package:dio/dio.dart';
+import 'package:http_mock_adapter/http_mock_adapter.dart';
+import 'package:requests_signature_dart/src/client/requests_signature_options.dart';
+import 'package:test/test.dart';
+import 'package:requests_signature_dart/requests_signature_dart.dart';
+
+void main() {
+  group('Requests Signature Interceptor Test', () {
+    final testClientId = 'test_client_id';
+    final testClientSecret = 'test_client_secret';
+    final headerName = 'X-Signature';
+    final signaturePattern = '{ClientId}:{Nonce}:{Timestamp}:{SignatureBody}';
+    final baseEndpoint = 'https://google.ca/';
+    final clockskewMS = 6000; // Clock skew in milliseconds
+    final toleranceMS = 500; // Tolerance in milliseconds
+    final emptyDataInjection = '{}'; //Injecting empty data as String
+
+    late Dio dio;
+    late DateTime now;
+
+    setUp(() {
+      dio = Dio();
+      now = DateTime.now().toUtc();
+    });
+
+    tearDown(() {
+      dio.close();
+    });
+
+    test('Interceptor adds signature header to GET request', () async {
+      // Arrange
+      final options = RequestsSignatureOptions(
+        clientId: testClientId,
+        clientSecret: testClientSecret,
+        headerName: headerName,
+        signaturePattern: signaturePattern,
+      );
+
+      final interceptor = RequestsSignatureInterceptor(options, dio);
+
+      dio.interceptors.add(interceptor);
+
+      // Act
+      final response = await dio.get(baseEndpoint);
+
+      // Assert
+      expect(response.statusCode, 200);
+
+      expect(response.requestOptions.headers, contains(headerName));
+    });
+
+    test('Interceptor returns OK with auto-retry tuned to false', () async {
```

Comment:
> Interceptor returns OK with auto-retry tuned to false

What is this testing?

When the auto-retry is false, it means that the interceptor must not adjust its timestamps using the clock skew. The result of this should be that you ultimately get an unauthorized response from the server.

To get an OK response, you would have to send a valid signature, which implies that you wouldn't need the auto-retry mechanism anyways.